mesonbuild/dependencies/dub.py

```diff
@@ -351,11 +352,11 @@ def _find_in_cache_db(self, spec: JsonDict, check_list: T.Set[str]) -> T.Tuple[s
             DubDependency.class_cache_dir = self._find_cache_dir()
         cache_dir = DubDependency.class_cache_dir
 
-        pkg = spec['name']
+        pkg = str(spec['name'])
```

Comment:
Instead of casting this to str because other spec entries could be a list, we can probably define JsonDict as a T.TypedDict instead.

Also please squash this change.

Reply:
I didn't know TypedDict. It saves so much trouble !